nix/packages/grub-efi-image.nix

```diff
@@ -0,0 +1,47 @@
+{ lib, stdenvNoCC, grub2_efi, fetchFromGitHub, git, gnulib }:
+let
+  grub_uki = grub2_efi.overrideAttrs (finalAttrs: previousAttrs: rec {
+    patches = [ ];
+    src = fetchFromGitHub {
+      owner = "GovanifY";
+      repo = "grub";
+      rev = "master";
+      sha256 = "sha256-2XhZdRCK73QJ5QIeTdqXVFinzXq36pJ/6EwnluY5rnA=";
+    };
+
+
+    prePatch = ''
+      patchShebangs autogen.sh bootstrap
+      # copy gnulib into build dir and make writable.
+      # Otherwise ./bootstrap copies the non-writable files from nix store and fails to modify them
+      cp -r ${gnulib} gnulib
+      chmod -R u+w gnulib/{build-aux,lib}
+      ./bootstrap --no-git --gnulib-srcdir=gnulib --skip-po
+      ./autogen.sh
+    '';
+    nativeBuildInputs = previousAttrs.nativeBuildInputs ++ [ git ];
+    preConfigure = "";
+
+
+  });
+in
+stdenvNoCC.mkDerivation rec {
+  pname = "grub-efi-image";
+  version = "unstable-uki";
+
+  buildCommand = ''
+    mkdir $out
+    cd /tmp
+    # this is very probably a terrible idea but grub doesn't allow to generate
+    # at runtime menus like systemd-boot does, so we will need to wait for
+    # lanzaboot to be ready to generate config for both sd-boot and grub.
+    echo -e 'set root=(hd0,gpt2)\nboot_loader_interface\nchainloader (hd0,gpt1)/efi/Linux/nixos-generation-1.efi\nboot' > /tmp/grub.cfg
+    ${grub_uki}/bin/grub-mkstandalone -O x86_64-efi -o $out/boot.efi \
```

Comment:
grub_uki should go to nativeBuildInputs otherwise cross compilation would not work.

Reply:
lanzaboote support for non x86 is still in PR AFAIK and grub-mkstandalone currently outputs for x86 efi only. This is not a simple matter of just changing something to native, it is more involved than this.

Reply:
I believe your misunderstanding his point. If grub-mkstandalone can output on non-x86 build machines, then it should be added to nativeBuildInputs so you can cross-compile from e.g. aarch64 to x86_64.

Reply:
Ah indeed I misunderstood, although both in cross-compilation and just efi targeting we currently don't support/target anything else than x86 because we don't have the means to test it (see https://github.com/nix-community/lanzaboote/pull/31 ).

I'll add it to nativeBuild but there is a lot more work to do than this I'm afraid. Thanks for the heads up!